File: Assets/Scripts/FullscreenSprite.js
#pragma strict

var chosenCamera : Camera;

function Awake() {
	var spriteRenderer = GetComponent.<SpriteRenderer>();

  var cameraHeight = chosenCamera.orthographicSize * 2;
  var cameraSize = Vector2(chosenCamera.aspect * cameraHeight, cameraHeight);
  var spriteSize = spriteRenderer.sprite.bounds.size;

	var scale = transform.localScale;
	Debug.Log(scale);
		Debug.Log(cameraSize.x);
		Debug.Log(cameraSize.y);

			Debug.Log(spriteSize.x);
			Debug.Log(spriteSize.y);
	if(cameraSize.x >= cameraSize.y) {
		scale *= cameraSize.x / spriteSize.x;
	} else {
		scale *= cameraSize.y / spriteSize.y;
	}
	Debug.Log(scale);

	transform.localScale = scale;
}

File: Assets/Scripts/FullscreenSprite.cs
using UnityEngine;
using System.Collections;

[System.Serializable]
public partial class FullscreenSprite : MonoBehaviour
{
    public Camera chosenCamera;
    public virtual void Awake()
    {
        SpriteRenderer spriteRenderer = this.GetComponent<SpriteRenderer>();
        float cameraHeight = this.chosenCamera.orthographicSize * 2;
        Vector2 cameraSize = new Vector2(this.chosenCamera.aspect * cameraHeight, cameraHeight);
        Vector3 spriteSize = spriteRenderer.sprite.bounds.size;
        Vector3 scale = this.transform.localScale;
        Debug.Log(scale);
        Debug.Log(cameraSize.x);
        Debug.Log(cameraSize.y);
        Debug.Log(spriteSize.x);
        Debug.Log(spriteSize.y);
        if (cameraSize.x >= cameraSize.y)
        {
            scale = scale * (cameraSize.x / spriteSize.x);
        }
        else
        {
            scale = scale * (cameraSize.y / spriteSize.y);
        }
        Debug.Log(scale);
        this.transform.localScale = scale;
    }

}